lib/httplog/http_log.rb

```diff
@@ -127,14 +133,25 @@ def parse_body(body, encoding, content_type)
         end
       end
 
-      utf_encoded(body.to_s, content_type)
+      result = utf_encoded(body.to_s, content_type)
+
+      if mask_body && body && !body.empty?
+        if content_type =~ /json/
+          begin
+            result = masked_data(config.json_parser.load(result))
+          rescue => e
+            e.message + ': ' + result
```

Comment:
This silently swallow a failure. Perhaps we should just log an error message and return an empty string so we're not accidentally leaking sensitive data into the logfile. Or looking further down, perhaps that's what's indented already and you forgot to re-raise the exception here?